PyMca5/PyMcaCore/TiledDataSource.py

```diff
@@ -0,0 +1,30 @@
+import copy
+
+from .DataObject import DataObject
+
+
+SOURCE_TYPE = "Tiled"
+
+
+class TiledDataSource(object):
+    """Manages a Tiled CatalogOfBlueskyRuns as a PyMca Data Source"""
+    def getDataObject(self, key: str, selection=None):
+        """Return data associated with 'key' and optional 'selection'."""
+        data = DataObject()
+        labels = ()
+        if selection is not None:
+            labels = selection["Channel List"].copy()
+        data.info = {
+            # For now, only allow one-dimensional data arrays
+            "selectiontype": "1D",
+            "selection": copy.deepcopy(selection),
+            "LabelNames": labels,
+        }
+        data.x = ()
```

Comment:
What does this look like when it is non-empty? Is a "record" with one or more data arrays, or is it an (N-D) array? The `tuple` caught my eye here.

Reply:
At the moment, we're only considering 1D data, so this will be a sequence of 1-D ndarrays.  Here's an [example of non-empty data](https://github.com/AbbyGi/pymca/pull/1/files#diff-1388fc547d704068040765224f5c11bb94389670d7dded61daf943168406b252R78-R81) from another branch of ours.

In that same branch I build in a [rudimentary enforcer of maximum array dimensions](https://github.com/AbbyGi/pymca/pull/1/files#diff-1388fc547d704068040765224f5c11bb94389670d7dded61daf943168406b252R117-R121), so that we could later expand this to 2D or 3D data.

The tuple is a personal preference that I prefer to use when the sequence should not be mutable.